src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Repositories/PolicyRepository.cs

```diff
@@ -11,62 +12,86 @@
 namespace Altinn.Platform.Authorization.Repositories
 {
     /// <summary>
-    /// Repository for handling authorization rules 
+    /// Repository for handling authorization rules
     /// </summary>
     public class PolicyRepository : IPolicyRepository
     {
+        private readonly ILogger<PolicyRepository> logger;
         private readonly AzureStorageConfiguration _storageConfig;
-        private readonly CloudBlobClient _blobClient;
-        private readonly CloudBlobContainer _blobContainer;
+        private CloudBlobClient _blobClient;
+        private CloudBlobContainer _blobContainer;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PolicyRepository"/> class
         /// </summary>
         /// <param name="storageConfig">The storage configuration for Azure Blob Storage.</param>
-        public PolicyRepository(IOptions<AzureStorageConfiguration> storageConfig)
+        /// <param name="logger">logger</param>
+        public PolicyRepository(
+            IOptions<AzureStorageConfiguration> storageConfig,
+            ILogger<PolicyRepository> logger)
         {
-            _storageConfig = storageConfig.Value;
-
-            // connect to azure blob storage
-            StorageCredentials storageCredentials = new StorageCredentials(_storageConfig.AccountName, _storageConfig.AccountKey);
-            CloudStorageAccount storageAccount = new CloudStorageAccount(storageCredentials, true);
+            this.logger = logger;
 
-            _blobClient = CreateBlobClient(storageCredentials, storageAccount);
-            _blobContainer = _blobClient.GetContainerReference(_storageConfig.StorageContainer);
+            _storageConfig = storageConfig.Value;
+            SetUpBlobConnection();
         }
 
         /// <inheritdoc />
-        public Task<Stream> GetPolicy(string filepath)
+        public async Task<Stream> GetPolicyAsync(string filepath)
         {
-            throw new NotImplementedException();
-        }
+            bool blobExists = await _blobContainer.GetBlockBlobReference(filepath).ExistsAsync();
```

Comment:
checking that is exists and the using the value seems a bit much. Could we simplify to only call GetBlockBlobReference once?